### ISSUE TITLE
KerasCV Object Detection tutorials are Deprecated! Please Help!

### ISSUE THREAD

**Issue by Paryavi** [keras-team/keras-cv#2046]
**Short Description**

Hi @ianstenbit and @jbischof and @fchollet and @LukeWood,

Thank you for working on the KerasCV library, I am an Electronics Engineer Ph.D. student and I am using KerasCV detection.

@ianstenbit said and I quote;
"BoxCOCOMetrics were created when we were TF-only and won't support Torch+JAX, so they are likely to be deprecated. my recommendation would be to use the PyCOCOCallback for metric evaluation."

If BoxCOCOMetrics is deprecated, it invalidates existing tutorials on object detection using RetinaNet and YoloV8 with Keras_CV. For classification, Keras alone suffices, so Keras_CV is mostly valuable for object detection tasks, and we don't have a best practice anymore.  Specifically, would you please provide a best practice to use PyCOCOCallback in the detection pipeline, for replacing the chunk of code below in the Object detection tutorials. We users can not train without proper callback design.

Callback Source; https://keras.io/guides/keras_cv/object_detection_keras_cv/  

a custom callback to achieve TPU compatibility:

class EvaluateCOCOMetricsCallback(keras.callbacks.Callback):
    def __init__(self, data):
        super().__init__()
        self.data = data
        self.metrics = keras_cv.metrics.BoxCOCOMetrics(
            bounding_box_format="xywh",
            # passing 1e9 ensures we never evaluate until
            # `metrics.result(force=True)` is
            # called.
            evaluate_freq=1e9,
        )

    def on_epoch_end(self, epoch, logs):
        self.metrics.reset_state()
        for batch in tqdm.tqdm(self.data):
            images, y_true = batch[0], batch[1]
            y_pred = self.model.predict(images, verbose=0)
            self.metrics.update_state(y_true, y_pred)

        metrics = self.metrics.result(force=True)
        logs.update(metrics)
        return logs

================================================
This is what I did to be able to train with KerasCV and my dataset, but I am not sure it is best practice;
from keras_cv.callbacks import PyCOCOCallback

def dict_to_tuple(inputs):
return inputs["images"], bounding_box.to_dense(
inputs["bounding_boxes"], max_boxes=32
)

train_ds = train_ds.map(dict_to_tuple, num_parallel_calls=tf.data.AUTOTUNE)
val_ds = val_ds.map(dict_to_tuple, num_parallel_calls=tf.data.AUTOTUNE)

train_ds = train_ds.prefetch(tf.data.AUTOTUNE)
val_ds = val_ds.prefetch(tf.data.AUTOTUNE)
callback = PyCOCOCallback(
validation_data=val_ds,
bounding_box_format="xywh",
)
yolo.fit(
train_ds,
validation_data=val_ds,
epochs=2,
callbacks=[callback],
)

**Papers**

**Existing Implementations**

**Other Information**

**Comment by ianstenbit** [comment]
Please see the usage of PyCOCOCallback in the updated guide: https://github.com/keras-team/keras-io/pull/1487/files

This update is not yet pushed to keras.io but should be in the coming weeks.

**Comment by Paryavi** [comment]
Hey @ianstenbit and @jbischof, 
Do you have a timeline for when https://github.com/keras-team/keras-io/pull/1487/files will be pushed to Keras-io?  
Same question for Yolo example using Keras core;
https://colab.research.google.com/drive/1qtAk8GRF1Nt_9TtkkLMPBkqUUiBA3bB2?usp=sharing#scrollTo=aeDfZ2bjtb_i
